### PR TITLE
fix: Trim gapic_metadata.json to appropriate services

### PR DESCRIPTION
**Pull request by jskeet**
We assume the first service that was generated has the "right"
protobuf service, and discard any that have different protobuf
packages to that one.

So for example, when generating KMS and IAM, KMS first so only the
KMS services appear in the metadata.

### CODE REVIEW

File: Google.Api.Generator/Generation/MetadataGenerator.cs [hunk 1]
```diff
@@ -32,19 +32,23 @@ internal static class MetadataGenerator
 
         internal static string GenerateGapicMetadataJson(List<ServiceDetails> allServiceDetails)
         {
+            // We assume that the first service has the correct proto package, and we can discard
+            // services in other packages. This means it doesn't matter (for gapic_metadata.json)
```
Comment (amanda-tarafa) [inline Google.Api.Generator/Generation/MetadataGenerator.cs:36]: nit: Is the last sentence incomplete?

**Review by amanda-tarafa — APPROVED** [review]
LGTM

File: Google.Api.Generator/Generation/MetadataGenerator.cs (hunk 1 above)
Comment (jskeet) [reply Google.Api.Generator/Generation/MetadataGenerator.cs:36]: Whoops, yes. Will fix.

**Review by viacheslav-rostovtsev — APPROVED** [review]
I wonder if it exposes us to order instability issues in bazel rules dependencies...
LGTM

**Comment by jskeet** [comment]
If it does, we were already in trouble as we'd be using the wrong proto package.

**Pull request merged**